cli/integration_tests/composable_config/composing-cache.t

```diff
@@ -0,0 +1,102 @@
+Setup
+  $ . ${TESTDIR}/../setup.sh
+  $ . ${TESTDIR}/setup.sh $(pwd) ./monorepo
+
+This test covers:
+# - `cache:false` in root, override `cache:true` in workspace
+# - `cache:true` in root, override to `cache:false` in workspace
+# - No `cache` config in root, override `cache:false` in workspace
+# - `cache:false` in root still works if workspace has no turbo.json
+
+# cache:false in root, override to cache:true in workspace
+  $ ${TURBO} run cached-task-1 --filter=cached > tmp.log
+  $ cat tmp.log
+  \xe2\x80\xa2 Packages in scope: cached (esc)
+  \xe2\x80\xa2 Running cached-task-1 in 1 packages (esc)
+  \xe2\x80\xa2 Remote caching disabled (esc)
+  cached:cached-task-1: cache miss, executing a6607ea0e6633b06
+  cached:cached-task-1: 
+  cached:cached-task-1: > cached-task-1
+  cached:cached-task-1: > echo 'cached-task-1' > out/foo.min.txt
+  cached:cached-task-1: 
+  
+   Tasks:    1 successful, 1 total
+  Cached:    0 cached, 1 total
+    Time:\s+[.0-9]+m?s  (re)
+  
+  $ HASH=$(cat tmp.log | grep -E "cached:cached-task-1.* executing .*" | awk '{print $5}')
+  $ echo $HASH
+  [a-z0-9]{16} (re)
+  $ tar -tf $TARGET_DIR/node_modules/.cache/turbo/$HASH.tar.zst;
```

Comment:
Can we use `--cache-dir` for these tests since many tests implicitly rely on that value?

Reply:
That's a good idea, (correct me if I'm wrong), we'd still have to do this dance to get the task hash. Is there any benefit to replacing`$TARGET_DIR/node_modules/.cache/turbo` with a custom value?

Reply:
The only reason is that it would make it possible to write a simple "check the result" helper function.

The "grab hash" thing is actually the thing I want to fix more than anything else, but we don't make that easy.

Reply:
I'll keep things the way they are, since they're already written, but I'll write new tests the way you suggest if applicable. E.g. #3778.